<a href="https://colab.research.google.com/github/KishorAlagappan/multifunctional-nlp-toolkit/blob/main/Translation_Magic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**NLP - Natural Language Processing **

* Welcome to the Multifunctional NLP Toolkit – a complete Natural Language Processing solution that simplifies language-related tasks.
* This project includes three core functions: translation to various languages, automatic spelling correction, and skill-based resume filtering.
* Delve into the world of NLP and experience the convenience of seamless communication and information handling.

**1. Translation Magic**

  Installation and Import

In [6]:
!pip install googletrans

In [7]:
!pip install deep-translator

In [8]:
import pandas as pd
from deep_translator import GoogleTranslator

Load Dataset

In [13]:
dataframe = pd.read_csv('English.csv')
dataframe

English words/sentences
0                                                     Hi.
1                                                    Run!
2                                                    Run!
3                                                    Who?
4                                                    Wow!
...                                                   ...
175616  Top-down economics never works, said Obama. "T...
175617  A carbon footprint is the amount of carbon dio...
175618  Death is something that we're often discourage...
175619  Since there are usually multiple websites on a...
175620  If someone who doesn't know your background sa...

[175621 rows x 1 columns]

Sample Text
  *  Here, our dataset has '175620' english words/ sentences, so now I am doing sampling.

In [14]:
sample_dataframe = dataframe.iloc[::1756]

sample_dataframe.reset_index(drop=True, inplace=True)

french_list = []
german_list = []
spanish_list = []

In [15]:
sample_dataframe

English words/sentences
0                                                  Hi.
1                                          We saw you.
2                                        I ate apples.
3                                       I am so sorry.
4                                       You're biased.
..                                                 ...
96   I saw some small animals running away in all d...
97   I've never been to Boston. However, I have bee...
98   The drug must go through clinical trials befor...
99   She spends a lot of time helping her children ...
100  I love trying out new things, so I always buy ...

[101 rows x 1 columns]

Translator object for each language
  * English (Existing Language)
  * German
  * French
  * Spanish

In [16]:
french_translator = GoogleTranslator(source='english', target='french')
german_translator = GoogleTranslator(source='english', target='german')
spanish_translator = GoogleTranslator(source='english', target='spanish')

In [18]:
for i in (sample_dataframe.iloc[:, 0]):

    french_list.append(french_translator.translate(i))
    german_list.append(german_translator.translate(i))
    spanish_list.append(spanish_translator.translate(i))

In [19]:
full_dataframe =  pd.DataFrame({'French':french_list, 'German':german_list, 'Spanish':spanish_list})

In [20]:
translated_dataframe = pd.concat([sample_dataframe.iloc[:, 0], full_dataframe], axis=1)

In [27]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_colwidth', None)

In [29]:
translated_dataframe.head(101)

English words/sentences  \
0                                                                                                                                                  Hi.   
1                                                                                                                                          We saw you.   
2                                                                                                                                        I ate apples.   
3                                                                                                                                       I am so sorry.   
4                                                                                                                                       You're biased.   
5                                                                                                                                      May I use this?   
6                                                                                                                                     He's on his way.   
7                                                                                                                                     That's too hard.   
8                                                                                                                                    He is in trouble.   
9                                                                                                                                    No one is amused.   
10                                                                                                                                   You're no singer.   
11                                                                                                                                  I really miss you.   
12                                                                                                                                  This is expensive.   
13                                                                                                                                 Do you like robots?   
14                                                                                                                                 I'm a tourist, too.   
15                                                                                                                                 Tom has a blue car.   
16                                                                                                                                Do you like to sing?   
17                                                                                                                                I will go if you go.   
18                                                                                                                                They rented a house.   
19                                                                                                                               Are you their mother?   
20                                                                                                                               I heard the question.   
21                                                                                                                               Please air the futon.   
22                                                                                                                               What is he aiming at?   
23                                                                                                                              He came into the room.   
24                                                                                                                              I wish I were younger.   
25                                                                                                                         